android armv7 is supported or not

## Description

### Current State
I would like to know if it is currently possible to compile and run on android armv7
<!-- Please shortly describe the bug you have found -->

### Expected

<!-- Please shortly describe what you expect instead -->

## Environment


- Hardware Architecture: armv7
- Operating system: Android

The following information is optional. Please provide them only if you have built from source.
- C++ Compiler version:
- CMake version:
- CMake flags: -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_AOT_RUNTIME=ON
- Boost version:

## Steps to Reproduce

<!-- Please provide as much information as necessary to reproduce the bug. -->

> Hi @morn-0 We fixed the building issues on the armv7l platform: https://github.com/WasmEdge/WasmEdge/pull/2573/files and planned to ship it in the 0.13.0 release. Please let me know if this fixes your issue. Thanks.


Failed to build in WameEdge-0.13.4 source code. 

NDK Version is 26b

```
vscode ➜ /workspaces/WasmEdge-0.13.4 $ cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_PACKAGE="TGZ" -DWASMEDGE_BUILD_AOT_RUNTIME=OFF -DCMAKE_SYSTEM_NAME=Android -DCMAKE_SYSTEM_VERSION=23 -DCMAKE_ANDROID_ARCH_ABI=armeabi-v7a -DCMAKE_ANDROID_NDK=$ANDROID_NDK_HOME -DCMAKE_ANDROID_STL_TYPE=c++_static
-- Android: Targeting API '23' with architecture 'arm', ABI 'armeabi-v7a', and processor 'armv7-a'
-- Android: Selected unified Clang toolchain
-- The C compiler identification is Clang 17.0.2
-- The CXX compiler identification is Clang 17.0.2
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
fatal: not a git repository (or any parent up to mount point /workspaces)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Looking for C++ include filesystem
-- Looking for C++ include filesystem - found
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Looking for mmap
-- Looking for mmap - found
-- Looking for C++ include pwd.h
-- Looking for C++ include pwd.h - found
-- Build spdlog: 1.11.0
-- Build type: Release
-- Configuring done
-- Generating done
-- Build files have been written to: /workspaces/WasmEdge-0.13.4/build
vscode ➜ /workspaces/WasmEdge-0.13.4 $ cmake --build build
[59/83] Building C object thirdparty/ggml/CMakeFiles/utilGgml.dir/ggml-alloc.c.o
FAILED: thirdparty/ggml/CMakeFiles/utilGgml.dir/ggml-alloc.c.o 
/opt/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=armv7-none-linux-androideabi23  -I/workspaces/WasmEdge-0.13.4/thirdparty/ggml -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -fexceptions -O3 -DNDEBUG -flto=thin -fPIC -Wall -Wextra -Werror -Wno-error=pedantic -Wno-psabi -Wno-c++20-designator -Wno-c99-extensions -Wno-covered-switch-default -Wno-documentation-unknown-command -Wno-error=nested-anon-types -Wno-error=old-style-cast -Wno-error=unused-command-line-argument -Wno-error=unknown-warning-option -Wno-ctad-maybe-unsupported -Wno-gnu-anonymous-struct -Wno-keyword-macro -Wno-language-extension-token -Wno-newline-eof -Wno-shadow-field-in-constructor -Wno-signed-enum-bitfield -Wno-switch-enum -Wno-undefined-func-template -Wno-error=shadow-field -Wno-reserved-identifier -Wno-unused-parameter -Wno-unused-variable -Wno-unused-but-set-variable -Wno-unused-function -Wno-missing-braces -MD -MT thirdparty/ggml/CMakeFiles/utilGgml.dir/ggml-alloc.c.o -MF thirdparty/ggml/CMakeFiles/utilGgml.dir/ggml-alloc.c.o.d -o thirdparty/ggml/CMakeFiles/utilGgml.dir/ggml-alloc.c.o -c /workspaces/WasmEdge-0.13.4/thirdparty/ggml/ggml-alloc.c
/workspaces/WasmEdge-0.13.4/thirdparty/ggml/ggml-alloc.c:264:45: error: implicit conversion from 'unsigned long long' to 'size_t' (aka 'unsigned int') changes value from 1099511627776 to 0 [-Werror,-Wconstant-conversion]
static const size_t MEASURE_MAX_SIZE  = 1ULL<<40; // 1 TB
                    ~~~~~~~~~~~~~~~~    ~~~~^~~~
1 error generated.
[61/83] Building CXX object lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o
FAILED: lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o 
/opt/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi23 -DSPDLOG_COMPILED_LIB -I/workspaces/WasmEdge-0.13.4/thirdparty -I/workspaces/WasmEdge-0.13.4/build/lib/system -I/workspaces/WasmEdge-0.13.4/build/include -I/workspaces/WasmEdge-0.13.4/include -I/workspaces/WasmEdge-0.13.4/build/_deps/spdlog-src/include -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -fexceptions -frtti -stdlib=libc++ -O3 -DNDEBUG -flto=thin -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Werror -Wno-error=pedantic -Wno-psabi -Wno-c++20-designator -Wno-c99-extensions -Wno-covered-switch-default -Wno-documentation-unknown-command -Wno-error=nested-anon-types -Wno-error=old-style-cast -Wno-error=unused-command-line-argument -Wno-error=unknown-warning-option -Wno-ctad-maybe-unsupported -Wno-gnu-anonymous-struct -Wno-keyword-macro -Wno-language-extension-token -Wno-newline-eof -Wno-shadow-field-in-constructor -Wno-signed-enum-bitfield -Wno-switch-enum -Wno-undefined-func-template -Wno-error=shadow-field -Wno-reserved-identifier -pthread -MD -MT lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o -MF lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o.d -o lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o -c /workspaces/WasmEdge-0.13.4/lib/host/wasi/inode-linux.cpp
/workspaces/WasmEdge-0.13.4/lib/host/wasi/inode-linux.cpp:613:23: error: call to 'fromFileType' is ambiguous
  Filestat.filetype = fromFileType(SysFStat.st_mode);
                      ^~~~~~~~~~~~
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:323:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(mode_t Mode) noexcept {
                                   ^
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:343:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(uint8_t Type) noexcept {
                                   ^
/workspaces/WasmEdge-0.13.4/lib/host/wasi/inode-linux.cpp:1374:10: error: call to 'fromFileType' is ambiguous
  return fromFileType(Stat->st_mode);
         ^~~~~~~~~~~~
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:323:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(mode_t Mode) noexcept {
                                   ^
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:343:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(uint8_t Type) noexcept {
                                   ^
2 errors generated.
[72/83] Building CXX object lib/vm/CMakeFiles/wasmedgeVM.dir/vm.cpp.o
ninja: build stopped: subcommand failed.
```

> Hi @hangedfish Please use the following patch to disable the ggml dependency:
> 
> [#2802 (comment)](https://github.com/WasmEdge/WasmEdge/issues/2802#issuecomment-1718681508)

Fail to build inode-linux.cpp

```
vscode ➜ /workspaces/WasmEdge-0.13.4 $ cmake --build build
[62/77] Building CXX object lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o
FAILED: lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o 
/opt/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi23 -DSPDLOG_COMPILED_LIB -I/workspaces/WasmEdge-0.13.4/thirdparty -I/workspaces/WasmEdge-0.13.4/build/lib/system -I/workspaces/WasmEdge-0.13.4/build/include -I/workspaces/WasmEdge-0.13.4/include -I/workspaces/WasmEdge-0.13.4/build/_deps/spdlog-src/include -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -fexceptions -frtti -stdlib=libc++ -O3 -DNDEBUG -flto=thin -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Werror -Wno-error=pedantic -Wno-psabi -Wno-c++20-designator -Wno-c99-extensions -Wno-covered-switch-default -Wno-documentation-unknown-command -Wno-error=nested-anon-types -Wno-error=old-style-cast -Wno-error=unused-command-line-argument -Wno-error=unknown-warning-option -Wno-ctad-maybe-unsupported -Wno-gnu-anonymous-struct -Wno-keyword-macro -Wno-language-extension-token -Wno-newline-eof -Wno-shadow-field-in-constructor -Wno-signed-enum-bitfield -Wno-switch-enum -Wno-undefined-func-template -Wno-error=shadow-field -Wno-reserved-identifier -pthread -MD -MT lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o -MF lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o.d -o lib/host/wasi/CMakeFiles/wasmedgeHostModuleWasi.dir/inode-linux.cpp.o -c /workspaces/WasmEdge-0.13.4/lib/host/wasi/inode-linux.cpp
/workspaces/WasmEdge-0.13.4/lib/host/wasi/inode-linux.cpp:613:23: error: call to 'fromFileType' is ambiguous
  Filestat.filetype = fromFileType(SysFStat.st_mode);
                      ^~~~~~~~~~~~
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:323:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(mode_t Mode) noexcept {
                                   ^
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:343:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(uint8_t Type) noexcept {
                                   ^
/workspaces/WasmEdge-0.13.4/lib/host/wasi/inode-linux.cpp:1374:10: error: call to 'fromFileType' is ambiguous
  return fromFileType(Stat->st_mode);
         ^~~~~~~~~~~~
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:323:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(mode_t Mode) noexcept {
                                   ^
/workspaces/WasmEdge-0.13.4/lib/host/wasi/linux.h:343:36: note: candidate function
inline constexpr __wasi_filetype_t fromFileType(uint8_t Type) noexcept {
                                   ^
2 errors generated.
[71/77] Building CXX object lib/vm/CMakeFiles/wasmedgeVM.dir/vm.cpp.o
ninja: build stopped: subcommand failed
```

--- a/lib/host/wasi/inode-linux.cpp
+++ b/lib/host/wasi/inode-linux.cpp
@@ -610,7 +610,7 @@ INode::pathFilestatGet(std::string Path,
 
   Filestat.dev = SysFStat.st_dev;
   Filestat.ino = SysFStat.st_ino;
-  Filestat.filetype = fromFileType(SysFStat.st_mode);
+  Filestat.filetype = fromFileType(static_cast<mode_t>(SysFStat.st_mode));
   Filestat.nlink = SysFStat.st_nlink;
   Filestat.size = SysFStat.st_size;
   Filestat.atim = fromTimespec(SysFStat.st_atim);
@@ -1371,7 +1371,7 @@ INode::sockGetPeerAddr(__wasi_address_family_t *AddressFamilyPtr,
 }
 
 __wasi_filetype_t INode::unsafeFiletype() const noexcept {
-  return fromFileType(Stat->st_mode);
+  return fromFileType(static_cast<mode_t>(Stat->st_mode));
 }
 
 WasiExpect<__wasi_filetype_t> INode::filetype() const noexcept {

